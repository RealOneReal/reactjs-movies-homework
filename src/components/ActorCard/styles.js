const styles = {
  card: {
    width: "150px",
    height: "225px",
    cursor: "pointer",
    ":hover ": {
      transform: "translateY(-16px)",
      transition: ".3s ease",
    },
    bgcolor: "secondary.dark",
  },
  cardMedia: { height: "170px", borderRadius: "10px" },
  cardContent: { p: "4px" },
};

export default styles;
