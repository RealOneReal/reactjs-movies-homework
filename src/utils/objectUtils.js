import { API_MOVIES } from "../api/moviesAPI";

export const renameKeysOfFetchingMovies = (obj) => {
  return {
    ...obj.data,
    results: obj.data.results?.map((item) => {
      const path = item.poster_path
        ? `${API_MOVIES.BASE_URL_IMAGES}${item.poster_path}`
        : "null";
      return {
        genreIds: item.genre_ids,
        title: item.title,
        posterPath: path,
        voteAverage: item.vote_average,
        id: item.id,
      };
    }),
  };
};
