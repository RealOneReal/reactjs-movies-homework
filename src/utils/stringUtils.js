export const truncateString = (str, number) => {
  if (str.length < number) return str;
  return str.slice(0, number - 4) + " ...";
};
