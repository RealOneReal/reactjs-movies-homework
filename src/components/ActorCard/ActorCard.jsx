import { Card, CardMedia, CardContent, Typography } from "@mui/material";
import styles from "./styles";
import { truncateString } from "../../utils/stringUtils";
import Fallback from "../../assets/broken-image.png";

const ActorCard = ({ id, profile_path, name, character }) => {
  const actorName = truncateString(name, 19);
  const characterName = truncateString(character, 19);

  const handleClick = () => {
    console.log(id);
  };

  const handleMediaFallback = (event) => {
    event.target.style.objectFit = "contain";
    return (event.target.src = Fallback);
  };

  return (
    <Card sx={styles.card} onClick={handleClick}>
      <CardMedia
        component="img"
        alt={actorName}
        src={profile_path}
        sx={styles.cardMedia}
        onError={handleMediaFallback}
      />
      <CardContent sx={styles.cardContent}>
        <Typography component="div" variant="subtitle1">
          {actorName}
        </Typography>
        <Typography component="div" variant="subtitle2">
          {characterName}
        </Typography>
      </CardContent>
    </Card>
  );
};

export default ActorCard;
