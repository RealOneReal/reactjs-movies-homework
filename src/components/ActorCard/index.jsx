import { Card, CardMedia, CardContent, Typography } from "@mui/material";
import { truncateString } from "../../utils/stringUtils";

const ActorCard = ({ id, profile_path, name, character }) => {
  const handleClick = () => {
    console.log(id);
  };

  const actorName = truncateString(name, 19);
  const characterName = truncateString(character, 19);

  const styles = {
    card: {
      width: "150px",
      height: "225px",
      cursor: "pointer",
      ":hover ": {
        transform: "translateY(-16px)",
        transition: ".3s ease",
      },
      bgcolor: "secondary.dark",
    },
    cardMedia: { height: "170px", borderRadius: "10px" },
  };

  return (
    <Card sx={styles.card} onClick={handleClick}>
      <CardMedia component="img" image={profile_path} sx={styles.cardMedia} />
      <CardContent sx={{ p: "4px" }}>
        <Typography component="div" variant="subtitle1">
          {actorName}
        </Typography>
        <Typography component="div" variant="subtitle2">
          {characterName}
        </Typography>
      </CardContent>
    </Card>
  );
};

export default ActorCard;
