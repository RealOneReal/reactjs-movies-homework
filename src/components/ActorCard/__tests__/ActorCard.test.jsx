import { render, screen, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import ActorCard from "../ActorCard";
import { truncateString } from "../../../utils/stringUtils";

const mockData = {
  id: 123,
  profile_path: "/123",
  name: "Michael Nordman",
  character: "Dude George",
};

jest.spyOn(window.console, "log");
jest.mock("../../../utils/stringUtils");
beforeEach(() => {
  truncateString
    .mockReturnValueOnce(mockData.name)
    .mockReturnValueOnce(mockData.character);
});

describe("ActorCard", () => {
  it("should be rendered", () => {
    const { container } = render(<ActorCard {...mockData} />);
    expect(container).toMatchSnapshot();
  });

  it("should be clickable", () => {
    render(<ActorCard {...mockData} />);
    const card = screen.getByAltText(/Michael Nordman/i);
    userEvent.click(card);
    expect(console.log).toBeCalled();
  });

  it("should have broken image at background on fallback", () => {
    render(<ActorCard {...mockData} />);
    const image = screen.getByAltText(/Michael Nordman/i);
    fireEvent.error(image);
    expect(image.src).toMatch(/broken-image.png$/i);
  });
});
