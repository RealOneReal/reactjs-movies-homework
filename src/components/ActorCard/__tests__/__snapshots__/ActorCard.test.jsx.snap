// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActorCard should be rendered 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1ifd40a-MuiPaper-root-MuiCard-root"
  >
    <img
      alt="Michael Nordman"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-1adevo0-MuiCardMedia-root"
      src="/123"
    />
    <div
      class="MuiCardContent-root css-102cj0y-MuiCardContent-root"
    >
      <div
        class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
      >
        Michael Nordman
      </div>
      <div
        class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
      >
        Dude George
      </div>
    </div>
  </div>
</div>
`;
